polaris.shopify.com/content/components/tables/index-table.md

```diff
@@ -180,31 +183,63 @@ An `IndexTableRow` is used to render a row representing an item within an `Index
+| scope?     | [HTMLTableCellElement['scope']](https://www.w3schools.com/tags/att_scope.asp)     | For subheader cells -- Indicates the cells that the `th` element relates to                                                                                              |
+| headers?   | [HTMLTableCellElement['headers']](https://www.w3schools.com/tags/att_headers.asp) | A space-separated list of the `th` cell IDs that describe or apply to it. Use for cells within a row that relate to a subheader cell in addition to their column header. |
 
 ---
 
 ## Related components
 
-- To create an actionable list of related items that link to details pages, such as a list of customers, use the [resource list component](https://polaris.shopify.com/components/resource-list)
-- To present structured data for comparison and analysis, like when helping merchants to gain insights or review analytics, use the [data table component](https://polaris.shopify.com/components/data-table)
+- To create an actionable list of related items that link to details pages, such as a list of customers, use the [resource list component](https://polaris.shopify.com/components/lists/resource-list)
+- To present structured data for comparison and analysis, like when helping merchants to gain insights or review analytics, use the [data table component](https://polaris.shopify.com/components/tables/data-table)
 - To display a simple list of related content, [use the list component](https://polaris.shopify.com/components/lists/list)
+
+---
+
+## Accessibility
+
+### Structure
+
+The `IndexTable` is an actionable, filterable, and sortable table widget that supports row selection with [subheaders](https://www.w3.org/WAI/tutorials/tables/multi-level/). To ensure that the power of this table is accessible to all merchants when implementing `IndexTable.Row` subheaders, set the following props on `IndexTable.Cell` that are appropriate for the enhancement you are implementing.
+
+Merchants can select a group of rows at once by clicking or <kbd>Space</kbd> keypressing a subheader row's checkbox. To indicate that an `IndexTable.Row` serves as a subheader for 1 or more rows below it, set the:
+
+- Zero-indexed table `position` of the first and last `IndexTable.Row` described by the subheader `IndexTable.Row` as a tuple array on its `subHeaderRange` prop
+- Unique `id` on the `IndexTable.Cell` that contains the subheader content
+- Element tag to `"th"` on the `as` prop of the subheader `IndexTable.Cell`
+- Subheader `IndexTable.Cell` `scope` prop to `"colgroup"`
+
+To associate the subheader `IndexTable.Row` with each `IndexTable.Cell` that it describes, set the:
+
+- Unique `id` provided to the subheader `IndexTable.Cell` on the `headers` prop of each related `IndexTable.Cell` (contained by an `IndexTable.Row` that's position is within the `subHeaderRange`) as well as the unique `id` of its corresponding column heading that you provided to the `IndexTable` `headings` prop
```

Comment:
☝️ This all sounds like a lot for an implementer to get right and maintain correctly.

Is there a better API here which doesn't require quite so much wiring up of `id` and setting `as`, etc?

At the very least, could we not expose a `IndexTable.SubHeader` component which renders `<Cell as="th"` and makes `subHeaderRange` a required prop?